Device/Device.cs

```diff
@@ -85,6 +91,11 @@ public async Task SubscribeToEventAsync(Action<MqttApplicationMessageReceivedEve
             await mqttClient.SubscribeAsync(topicC2D, MQTTnet.Protocol.MqttQualityOfServiceLevel.AtLeastOnce);
         }
 
+        public void DisconnectUngracefully()

```

Comment:
Why ungracefully? :D

Reply:
To test the `WILL` message it is required to shutdown the client unexpectedly, hence the "Ungracefully" I guess in the method name.

Reply:
Yes, exactly. I know it sounds old-fashioned :D, but found it in this link https://www.hivemq.com/blog/mqtt-essentials-part-9-last-will-and-testament/